cpp/src/table/row_operators.cu

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/structs/utilities.hpp>
+#include <cudf/detail/utilities/vector_factories.hpp>
+#include <cudf/table/experimental/row_operators.cuh>
+#include <cudf/table/table_view.hpp>
+
+#include <jit/type.hpp>
+
+namespace cudf {
+namespace experimental {
+
+namespace {
+
+auto struct_lex_verticalize(table_view table,
+                            host_span<order const> column_order         = {},
+                            host_span<null_order const> null_precedence = {})
+{
+  std::vector<column_view> verticalized_columns;
+  std::vector<order> new_column_order;
+  std::vector<null_order> new_null_precedence;
```

Comment:
```suggestion
  std::vector<order> verticalized_column_order;
  std::vector<null_order> verticalized_null_precedence;
```
Does the above remaing make sense?

Reply:
I didn't name it that because I thought that the new `verticalized_column_order` is the column order of the verticalized columns but it is not verticalized itself. But reading the name, it seemed to imply the latter.

Reply:
I found it's more consistent with `verticalized_columns` and `verticalized_col_depths` but doesn't matter. I'm closing this issue.